Src/Particle/AMReX_ParticleInit.H

```diff
@@ -1004,7 +1004,7 @@ InitRandom (Long                    icount,
                         r = amrex::Random();
                         x = geom.ProbLo(i) + (r * len[i]);
                     }
-                    while (x < xlo[i] || x > xhi[i]);
+                    while (static_cast<ParticleReal>(x) < static_cast<ParticleReal>(xlo[i]) || static_cast<ParticleReal>(x) >= static_cast<ParticleReal>(xhi[i]));
```

Comment:
That's confusing, `x`, `xlo` and `xhi` are all of `Real` type for this comparison, no? Why do we need the casts here, too?

Reply:
Because later we assign the double precision x to a single precision variable.

Reply:
I see.
I think since casts are expensive, I would in that case store `x`, `xhi` and `xlo` as `ParticleReal` and cast on assignment.

Reply:
I am hoping the compiler could figure it out.

Reply:
I fear they usually do not :-S